src/bandersnatch_storage_plugins/filesystem.py

```diff
@@ -97,7 +97,7 @@ def rewrite(
         self.delete_file(filepath_tmp)
 
     @contextlib.contextmanager
-    def update_safe(self, filename: str, **kw: Any) -> Generator[IO, None, None]:
+    def update_safe(self, filename: PATH_TYPES, **kw: Any) -> Generator[IO, None, None]:
```

Comment:
@techalchemy - This should be legit - my quick looks seems both can be handled or am I missing something?

Reply:
Dan ok'd in IRC. Merging :D